core/src/main/java/io/grpc/CallOptions.java

```diff
@@ -104,6 +127,7 @@ public String toString() {
       buffer.append(" (").append(deadlineNanoTime - System.nanoTime()).append(" ns from now)");
     }
     buffer.append("]");
+    buffer.append("\ncodec=").append(codec);
```

Comment:
We also use Preconditions extensively.  If we care about breaking the dep, we can clone the library and cherry pick in functions.